src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Node.cs

```diff
@@ -0,0 +1,160 @@
+            }
+
+            return ReadUsingMetadata<TValue>(node, jsonTypeInfo);
+        }
+
+        /// <summary>
+        /// Converts the <see cref="JsonNode"/> representing a single JSON value into a <paramref name="returnType"/>.
+        /// </summary>
+        /// <returns>A <paramref name="returnType"/> representation of the JSON value.</returns>
+        /// <param name="node">The <see cref="JsonNode"/> to convert.</param>
+        /// <param name="returnType">The type of the object to convert to and return.</param>
+        /// <param name="context">A metadata provider for serializable types.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// <paramref name="returnType"/> is <see langword="null"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="context"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="JsonException">
+        /// The JSON is invalid.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="returnType" /> is not compatible with the JSON.
+        ///
+        /// -or-
+        ///
+        /// There is remaining data in the string beyond a single JSON value.</exception>
+        /// <exception cref="NotSupportedException">
+        /// There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/>
+        /// for <paramref name="returnType"/> or its serializable members.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// The <see cref="JsonSerializerContext.GetTypeInfo(Type)"/> method of the provided
+        /// <paramref name="context"/> returns <see langword="null"/> for the type to convert.
+        /// </exception>
+        public static object? Deserialize(this JsonNode? node, Type returnType, JsonSerializerContext context)
+        {
+            if (returnType == null)
+            {
+                throw new ArgumentNullException(nameof(returnType));
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            return ReadUsingMetadata<object?>(node, GetTypeInfo(context, returnType));
+        }
+
+        [RequiresUnreferencedCode(SerializationUnreferencedCodeMessage)]
+        private static TValue? ReadUsingOptions<TValue>(JsonNode? node, Type returnType, JsonSerializerOptions? options) =>
+            ReadUsingMetadata<TValue>(node, GetTypeInfo(returnType, options));
+
+        private static TValue? ReadUsingMetadata<TValue>(JsonNode? node, JsonTypeInfo jsonTypeInfo)
+        {
+            JsonSerializerOptions options = jsonTypeInfo.Options;
+            Debug.Assert(options != null);
+
+            // For performance, we avoid calling JsonSerializer.Deserialize(node) since that will allocate an extra byte[].
+
+            using (var output = new PooledByteBufferWriter(options.DefaultBufferSize))
```

Comment:
Nit: could avoid two layers of indentation with `using var ` syntax.

Reply:
I did clean up the `PooledByteBufferWriter`, but left the writer's `using` since it isn't used all the way to the end of the method and I prefer the variable scoping.